### PR TITLE
Logbook: filter by entity and period

### PR DESCRIPTION
**Pull request by Anonym-tsk**
## Description:

Support filtering logbook by entity and period.
PR to backend: https://github.com/home-assistant/home-assistant/pull/17095

Preview

### CODE REVIEW

File: src/panels/logbook/ha-logbook-data.js [hunk 1]
```diff
@@ -33,41 +45,71 @@ class HaLogbookData extends PolymerElement {
...
     if (!this.hass) return;
 
     this._setIsLoading(true);
 
-    this.getDate(filterDate).then(function (logbookEntries) {
-      this._setEntries(logbookEntries);
-      this._setIsLoading(false);
-    }.bind(this));
+    this.getDate(this.filterDate, this.filterPeriod, this.filterEntity)
+      .then(function (logbookEntries) {
```
Comment (balloob) [inline src/panels/logbook/ha-logbook-data.js]: Don't use bind. Instead use an arrow function which takes the context of the location it's defined: `logbookEntries => { … }`

File: src/panels/logbook/ha-logbook-data.js [hunk 2]
```diff
@@ -33,41 +45,71 @@ class HaLogbookData extends PolymerElement {
...
-          logbookEntries.reverse();
-          return logbookEntries;
-        },
-        function () {
-          DATE_CACHE[date] = false;
-          return null;
-        }
-      );
+  getDate(date, period, entityId) {
+    if (!period) period = DEFAULT_PERIOD;
```
Comment (balloob) [inline src/panels/logbook/ha-logbook-data.js]: The data element should not handle defaults, defaults should be handled by the business logic that controls this element.

Comment (balloob) [reply src/panels/logbook/ha-logbook-data.js]: On a side note, we should just kill this element and make it a JavaScript function.

File: src/panels/logbook/ha-logbook.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ class HaLogbook extends EventsMixin(PolymerElement) {
 
     <template is="dom-repeat" items="[[entries]]">
       <div class="horizontal layout entry">
-        <div class="time">[[_formatTime(item.when)]]</div>
+        <div class="date">[[_formatDateTime(item.when)]]</div>
```
Comment (balloob) [inline src/panels/logbook/ha-logbook.js]: This makes for a very messy UI. Too much data that is not relevant. I think that we should never show a date if we're just showing a single day. We also should consider only showing a new date heading when the next entry is on another day than the previous one.

File: src/panels/logbook/ha-panel-logbook.js [hunk 4]
```diff
@@ -75,14 +103,38 @@ class HaPanelLogbook extends LocalizeMixin(PolymerElement) {
...
+              <paper-item>[[localize('ui.duration.week', 'count', 1)]]</paper-item>
+            </paper-listbox>
+          </paper-dropdown-menu>
+  
+          <ha-entity-picker
+            hass="[[hass]]"
+            value="{{_entityId}}"
+            label="[[localize('ui.components.entity.entity-picker.entity')]]"
+            disabled='[[isLoading]]'
+            onChange="{{entityPicked}}"
```
Comment (balloob) [inline src/panels/logbook/ha-panel-logbook.js]: `on-change`

File: src/panels/logbook/ha-panel-logbook.js [hunk 5]
```diff
@@ -22,6 +23,11 @@ import LocalizeMixin from '../../mixins/localize-mixin.js';
  * @appliesMixin LocalizeMixin
  */
 class HaPanelLogbook extends LocalizeMixin(PolymerElement) {
+  constructor() {
+    super();
+    this.entityPicked = this._entityPicked.bind(this);
```
Comment (balloob) [inline src/panels/logbook/ha-panel-logbook.js]: This is not necessary with Polymer event listeners.

File: src/panels/logbook/ha-panel-logbook.js (hunk 4 above)
Comment (Anonym-tsk) [reply src/panels/logbook/ha-panel-logbook.js]: No, HaEntityPicker works only with onChange

File: src/panels/logbook/ha-panel-logbook.js (hunk 5 above)
Comment (Anonym-tsk) [reply src/panels/logbook/ha-panel-logbook.js]: Otherwise we lose context.

File: src/panels/logbook/ha-panel-logbook.js (hunk 4 above)
Comment (balloob) [reply src/panels/logbook/ha-panel-logbook.js]: You must have looked at the Preact components. https://github.com/home-assistant/home-assistant-polymer/blob/master/src/panels/dev-service/ha-panel-dev-service.js#L110-L117

File: src/panels/logbook/ha-panel-logbook.js (hunk 5 above)
Comment (balloob) [reply src/panels/logbook/ha-panel-logbook.js]: Use `on-change` and you don't need it.

File: src/panels/logbook/ha-logbook-data.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { PolymerElement } from '@polymer/polymer/polymer-element.js';
 
-var DATE_CACHE = {};
+var DATA_CACHE = {};
```
Comment (balloob) [inline src/panels/logbook/ha-logbook-data.js]: Please use `const` instead of `var`

**Comment by balloob** [comment]
After final comment addressed and Travis build passes, this can be merged when the backend PR has been merged.

**Comment by Anonym-tsk** [comment]
@balloob There is one new not translated string with key 'ui.panel.logbook.period'. Please add it to translation system.

**Comment by balloob** [comment]
It will be added to the translation system automatically when this PR is merged.

**Review by balloob — APPROVED** [review]
This is ok to merge when backend is merged 👍

**Pull request merged**